src/main/scala/deduplication/Deduplication.scala

```diff
@@ -6,26 +6,63 @@ import java.util.concurrent.TimeoutException
 
 import scala.concurrent.duration._
 import scala.compat.java8.DurationConverters._
-import scala.collection.JavaConverters._
 
 import cats._
 import cats.implicits._
 import cats.effect._
-import cats.effect.implicits._
 
-import com.amazonaws.services.dynamodbv2.{AmazonDynamoDBAsync, AmazonDynamoDBAsyncClientBuilder}
-import com.amazonaws.services.dynamodbv2.model._
-import com.amazonaws.handlers._
-import org.scanamo._
-import org.scanamo.error.{DynamoReadError, NoPropertyOfType}
+import com.ovoenergy.comms.deduplication.model._
+import Deduplication._
+import io.chrisdavenport.log4cats.slf4j.Slf4jLogger
+import io.chrisdavenport.log4cats.StructuredLogger
 
-import model._
-import ScanamoHelpers._
+object Deduplication {
 
-/**
-  *
-  */
-trait Deduplication[F[_], ID] {
+  def nowF[F[_]: Functor: Clock] =
+    Clock[F]
+      .realTime(TimeUnit.MILLISECONDS)
+      .map(Instant.ofEpochMilli)
+
+  def processStatus[F[_]: Monad: Clock](
+      maxProcessingTime: FiniteDuration
+  )(p: Process[_, _]): F[ProcessStatus] = nowF[F].map { now =>
+
+    val isCompleted =
+      p.completedAt.isDefined
+
+    val isExpired = p.expiresOn
+      .map(_.instant)
+      .exists(_.isBefore(now))
```

Comment:
small thing, this can be `.map(_.instant.isBefore(now)).isDefined`, slightly more readable imho

Reply:
Uhm I though using exist would be more readable: 
> exists a case where the thing is before now